2009/misc/inputs/D-generate.c

```diff
@@ -0,0 +1,44 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#define TAM 100000000
+
+int main(int argc, char* argv[])
+{
+    char str[8];
+    long int tam;
+    int c;
+    FILE* fout;
+
+    srand(time(NULL));
+    fout = fopen("D.in", "w");
+    if (fout == NULL)
+    {
+        perror("fopen fout");
+        exit(EXIT_FAILURE);
+    }
+
+    if (argc == 1)
+        tam = TAM;
+    else
+        tam = atol(argv[1]);
+
+    fprintf(fout, "%ld\n", tam);
+
+    while (tam > 0)
+    {
+        for (c = 0; c < 7; c++)
+        {
+            str[c] = 33 + (rand() % 94);
+        }
+        str[c] = 0;
+        fprintf(fout, "%s\n", str);
+        tam--;
+    }
+
+    fclose(fout);
+
+    return EXIT_SUCCESS;

```

Comment:
How should we proceed with this kind of problems (where it is given a script to generate inputs)? I think we could provide an input and place it at the inputs folder, for automation proposes. We need to discuss a structure to handle this situation. Should we open an issue to talk more about that?

Reply:
I thought of using it to generate a static input file, BUT the input file that this script generates by default has ~700+ MB

I am planning to have 3 different ways of describing inputs in the repository:
* Normal input files: `X.in` files, small in size, direct to the point
* Zipped input files: `X.zip` files, for input files too great in size and without a script to generate it (it can be a temporary solution, until we develop scripts to generate them)
* Script-generated input files: `X-generate.c`. After compiled, it can (by default) generate a large input file, but also should have the ability to be parametrized in order to generate smaller inputs

What do you think?